lib/src/metta/runner/mod.rs

```diff
@@ -396,6 +397,12 @@ impl Metta {
         modules.get(mod_id.0).unwrap().space().clone()
     }
 
+    /// Returns a buffer containing the specified resource, if it is available from a loaded module
+    pub fn get_module_resource(&self, mod_id: ModId, res_key: &str) -> Result<Vec<u8>, String> {
+        let modules = self.0.modules.lock().unwrap();
+        modules.get(mod_id.0).unwrap().get_resource(res_key).clone()
```

Comment:
Cloning is redundant here:
```suggestion
        modules.get(mod_id.0).unwrap().get_resource(res_key)
```

Reply:
Good catch!

I had an API that initially returned a borrowed value, then I changed it but didn't change the clone.  I think the compiler would have optimized it away, but good catch.